File: Leetcode/300. Longest Increasing Subsequence/myanswer.c
int lengthOfLIS(int* nums, int numsSize){
    int lis=1;
    int lastmin[1000005]={-1000000001};

    for(int i=0;i<numsSize;i++){
        int a = nums[i];
        bool same=false;
        int left=0,right=lis-1,mid;
        if(lis==0){
            lastmin[lis]=a;
            lis++;
        }
        else if(lastmin[lis-1]<a){
            lastmin[lis]=a;
            lis++;
        }
        else{
            while(left<=right){
                mid=(left+right)/2;
                if(lastmin[mid]>a){
                    right=mid-1;
                }
                else if(lastmin[mid]==a){
                   same=true;
                    break;
                }
                else{
                    if(lastmin[mid+1]>a){
                        break;
                    }
                    left=mid+1;
                }
            }
            if(!same){
                lastmin[mid+1]=a;
            }
        }
    }
    return lis-1;
}

File: Leetcode/35. Search Insert Position/myanswer.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left=0
        right=len(nums)-1
        if(nums[left]>=target):
            return 0
        elif(nums[right]<target):
            return right+1
        while(left<=right):
            mid = (left+right)//2
            if(nums[mid]<target and target<=nums[mid+1]):
                return mid+1
            elif(nums[mid]==target):
                return mid
            elif(nums[mid]>target):
                right=mid
            elif(nums[mid+1]<target):
                left=mid+1
        return left
